Source/WebCore/page/scrolling/ScrollingTreeScrollingNode.cpp

```diff
@@ -172,10 +172,11 @@ FloatPoint ScrollingTreeScrollingNode::maximumScrollPosition() const
 
 bool ScrollingTreeScrollingNode::eventCanScrollContents(const PlatformWheelEvent& wheelEvent) const
 {
-#if PLATFORM(GTK)
-    // In case of GTK platform the end of momentum scroll events
-    // always have a delta of 0. More information in the documentation
-    // of the API gdk_event_is_scroll_stop_event.
+#if PLATFORM(WPE) || PLATFORM(GTK)
+    // The end of momentum scroll event can scroll contents even if with delta of 0:
+    // - For GTK, more information in the documentation of the API gdk_event_is_scroll_stop_event.
+    // - For WPE (libwpe), an axis motion event with a value of zero is treat as a 'stop' event in
+    //   the implementation of 'handle_pointer_event()' in Source/WebKit/UIProcess/API/wpe/WPEView.cpp.
```

Comment:
I would simplify this with just "In case of GTK and WPE the end of momentum scroll events always have a dela of 0." No need to point out to further info in any case.

Reply:
I don't mind For the case of GTK  but for the particular case of WPE I have doubts. It took to me time to find why this was having this behavior and this is artificially added in the code (https://github.com/WebKit/WebKit/blob/main/Source/WebKit/UIProcess/API/wpe/WPEView.cpp#L189) it doesn't come from any spec or protocol definition.


Having say that maybe it is enough just with the explanation done in the commit message.